src/components/elements/SkipToContentButton.tsx

```diff
@@ -0,0 +1,55 @@
+import { Button } from '@mui/material';
+import { ReactNode } from 'react';
+
+type SkipToContentProps = {
+  focusTargetId: string;
+  children?: ReactNode;
+};
+
+const SkipToContent: React.FC<SkipToContentProps> = ({
+  focusTargetId,
+  children,
+}) => {
+  const handleClick = () => {
+    const container = document.getElementById(focusTargetId);
+    if (container) {
+      // First ensure the container is focusable
+      if (!container.hasAttribute('tabindex')) {
+        container.setAttribute('tabindex', '-1');
+      }
+      container.focus();
+    }
+  };
+
+  return (
+    <Button
+      variant='contained'
+      onClick={handleClick}
+      sx={{
+        // Use `clip` to hide the button when it isn't focused, without preventing focus entirely (which display:none and visibility:hidden would do)
+        // `clip` is considered deprecated, but more widely supported than the more modern `clipPath`
+        clip: 'rect(0 0 0 0)',
+        clipPath: 'inset(50%)',
+
+        // Remove padding, border, and margin
+        padding: 0,
+        margin: 0,
+        border: 'none',
+        position: 'absolute', // Ensure it doesn't take up any space
+
+        '&:focus': {
+          // On focus, restore clip and other styles so the button is visible and styled normally
+          clip: 'auto',
+          clipPath: 'none',
+          position: 'relative',
```

Comment:
To discuss: Would it be better to give the button absolute positioning? Maybe optionally, as a prop, in some cases, such as in the app bar? (See e.g. the MUI docs Skip to Content button.)

Reply:
I'm not seeing a skip to content button appear on the MUI docs (eg https://mui.com/material-ui/getting-started/) when I tab the page. Or are you referencing a specific piece of documentation about how to implement Skip to Content?

Reply:
This is what I see when I open https://mui.com/material-ui/getting-started/ and immediately hit "tab":
![Screenshot 2025-01-06 at 4 58 44 PM](https://github.com/user-attachments/assets/7f2cf083-ee42-4dbf-a2a8-acd69cca2f18)
It has absolute positioning at the top left.

Reply:
I see. I have no preference here I think they way you implemented it works